code/modules/reagents/chemistry/machinery/smoke_machine.dm

```diff
@@ -0,0 +1,119 @@
+/obj/machinery/smoke_machine
+	name = "Smoke Machine"
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "smoke0"
+	density = TRUE
+	anchored = TRUE
+	circuit = /obj/item/circuitboard/machine/smoke_machine
+	var/efficiency = 10
+	var/on = FALSE
+	var/cooldown = 0
+	var/screen = "home"
+	var/useramount = 30 // Last used amount
+	var/volume = 1000
+	var/setting = 3
+	var/list/possible_settings = list(3,6,9,12,15)
+
+/datum/effect_system/smoke_spread/chem/smoke_machine/set_up(datum/reagents/carry = null, setting = 3, efficiency = 10, loc)
+	amount = setting
```

Comment:
shouldn't you call ..() somewhere?

Reply:
This is where I realize I copy pasted the code, go "uh sure", add it, then it crashes the servers and deletes everyone's preferences when it gets merged.

Reply:
Actually I'm not going to call it, just double checked and there's wayyy too much shit from the parents I don't want going on. This is the "no-nonsense" version of the parent's proc and like 80% of the reason I had to create a new smoke subtype for this.

Off the top of my head the parent would cause this machine to message admins about the effect every ~20 seconds.